homeassistant/components/onvif/camera.py

```diff
@@ -204,8 +204,10 @@ async def _async_get_stream_uri(self) -> str:
                 self._stream_uri_future.set_exception(err)
             raise
         url = URL(uri_no_auth)
-        url = url.with_user(self.device.username)
-        url = url.with_password(self.device.password)
+        if self.device.username:
```

Comment:
These methods both support `None` as an argument to clear the field. How about something like this?

```
url = url.with_user(self.device.username or None)
url = url.with_password(self.device.password or None)
```

Reply:
@jterrace, sounds good. Let me give it a spin.

Reply:
Works like a charm, applied. Thank you.